torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -361,6 +361,10 @@ def rotate_bounding_box(
     expand: bool = False,
     center: Optional[List[float]] = None,
 ) -> torch.Tensor:
+    if center is not None and expand:
+        warnings.warn("The provided center argument is ignored if expand is True")
+        center = None
```

Comment:
Given that both are not the default values, should we maybe fail if someone actually does this?

Reply:
It is not an error, that is why I raise a warning...

Reply:
It is not an error because we chose to ignore user input. If someone actually does `rotate_bounding_box(..., expand=True, center=[...])` I think we should just fail. Is there any use case for allowing both to be set?

Reply:
We can always imagine a use-case like data aug policy search where `expand` and `center` are parameters. Neither PIL nor D2 does not error in such case. I'm not sure why we should be that strict.

Reply:
> We can always imagine a use-case like data aug policy search where `expand` and `center` are parameters.

We should definitely not emit warnings from code where the user has no influence on. Imagine doing

```py
transform = transforms.NewAutoAug()
transform(image)
```

and getting a warning that the parameter `center` is ignored. Thus, internally `NewAutoAug` needs to do something like `center=center if not expand else None` anyway. 

> Neither PIL nor D2 does not error in such case. I'm not sure why we should be that strict.

1. What is D2?
2. Do they warn in this case?

Reply:
D2 = detectron2. 

> Do they warn in this case?

No, both are silent. Only docs states the behaviour.

Reply:
I just tried with PIL and it seems `center` has an effect on the output with `expand=True`

```py
image = PIL.Image.open("test/assets/encode_jpeg/grace_hopper_517x606.jpg")

image.rotate(30.0, center=(100, 100), expand=True).show()
image.rotate(30.0, center=(200, 200), expand=True).show()
```

![1](https://user-images.githubusercontent.com/6849766/160622574-12322325-07f5-47eb-b4e1-06ef40b8cb21.jpg) ![2](https://user-images.githubusercontent.com/6849766/160622678-696fcc5b-a2cf-4e4b-a39c-e6677985ad62.jpg)

Reply:
Yes, I should have mentioned that PIL modifies image but it is incorrect behaviour, IMO.

According to their docs:

> :param expand: Optional expansion flag.  If true, expands the output
>           image to make it large enough to hold the entire rotated image.
>           If false or omitted, make the output image the same size as the
>           input image.  Note that the expand flag assumes rotation around
>           the center and no translation.

https://github.com/python-pillow/Pillow/blob/a6efaa1ae1d0bc83bf1cbbe7170f7734a7f4203d/src/PIL/Image.py#L2122-L2123

Reply:
TorchVision has historically had a mixed approach on that. Sometimes we throw an error, sometimes we overwrite the value and emit a warning. Lately on models I started throwing errors when a parameter is incompatible. Might be worth aligning all places in TorchVision on follow up PRs but I'll leave it to you to decide how to handle this in this case; no strong opinions.

Reply:
Maybe we can rephrase the warning to make it more clear what is happening? The passed `center` is not ignored, but rather has no effect on the result if `expand=True`. This makes it more clear that we are not ignoring user input, but rather warn the user that the input combination makes little sense since it doesn't have an effect.